### DISCUSSION TITLE
Welcome to Z80-CPU-Tester Discussions!

### DISCUSSION THREAD

**Discussion by slabbi** [slabbi/Z80-CPU-Tester#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by berger1920** [comment]
Hello,

First of all, thank you so much for this new Z80 tester: of all the ones I've seen so far, it's the one I like the most. I say this without undervaluing the rest, as they all have their qualities.

Well, I was wondering about something that is mentioned in the manual but is not reflected on the PCB or in the Schematic. I am referring to the "Current measurement" jumper.

Has it been removed or is the manual more up to date than the gerber files and the schematic?

Best,
Gerardo "Berger"

**Comment by slabbi** [comment]
Indeed, it was on the prototype board, but the first feedback I got was, that user did not know how to use it.
They connected a voltmeter(!) or left the jumper open and wondered why it didn't work. So I removed it in rev. 1.1.

Currently I am thinking to add it again in the next release or to create a second board with more connectors for "proficient" users and keep this as an easy to use version, since it covers what most users want I guess.

Manuals sometimes lag a bit behind. Writing manuals is more work than creating a circuit diagram ;)

**Comment by berger1920** [comment]
Indeed it is. And even if it's not more work, it's more fun to make the circuit, than not to document everything.  :D

I thought that maybe someone had damaged the Z80 by manipulating the jumper in operation when connecting the ammeter (I don't know if this can happen), rather than by not knowing how to measure current.  :)

I'm designing my own PCB from your schematic, because I enjoy doing it (it really relaxes me to do the layout), so I will include that jumper next to the ZIF socket. Any other recommendations I can add for "expert" or future features?  ;)

**Comment by slabbi** [comment]
Not yet, I have some ideas but the software is more important.

When I create a new PCB layout it should have more features (without being a single board computer, it should still be simple and cheap to build). Adding a second connector is not very creative, so I prefer to improve the software first (more tests would be great, at least every opcode should be executed once).

**Comment by berger1920** [comment]
That sounds great. When I get my PCB assembled, I'll check all the Z80s I have, and I'll share the results. Thank you very much for your great work.  :)

**Comment by slabbi** [comment]
Great. Would be interesting to see how reliable it is with CPU’s from other sources.

**Comment by berger1920** [comment]
Hi Stephan,

If someday you make a revision of this design, maybe it would be a good idea to use some non-obsolete ICs like the 74ALS990 and the 74LS574. Fortunately I have them, because I've been working in electronics for many years and I keep a good stock of old chips, but if someone younger wanted to assemble this Z80 Tester, I think he/she would have problems to get those chips. What do you think?

**Comment by slabbi** [comment]
Do you have any good replacements for these?

**Comment by berger1920** [comment]
Hummm... For the 74LS574 I think it would be fine to use the 74HCT574 instead, Texas Instruments still produce it, it's TTL compatible, and it can supply enough current for the LEDs since 1K resistors are used. As for the 74ALS990, right now I don't know what replacement could be used.

**Comment by slabbi** [comment]
Then I see no problem.

See the notes on
https://github.com/slabbi/Z80-CPU-Tester/tree/main/Model%201/Hardware

_The 74LS574 can be LS or HCT (HC should work too)._

The 74ALS990 should be still active.

**Comment by berger1920** [comment]
Wow, I apologise. I had read that note on the first day, but then I forgot about it. :(

In any case, you are right, with that IC there is no problem, but there is with the 74ALS990, and that's why I wrote my comment... erroneously including the other one. Let's see:

**74ALS990**

Mouser: No Availability (97 Expected 8/23/2024).
Digikey: 0 stock.
TME: Not listed.
RS: Not listed.
Farnell: Not listed.
Reichelt: Not listed.

These are the suppliers I usually source my components from, I'm not saying there aren't any others. You can find some on eBay, but at high prices. For example here: [Kessler-electronic](https://www.ebay.de/str/kesslerelectronic)

https://www.ebay.de/itm/310655532474

**Comment by slabbi** [comment]
Ich lass mir was einfallen ;)

**Comment by berger1920** [comment]
I'm sorry. I just realized that I shouldn't have posted in this thread about the chip issue, but instead opened another discussion.
